src/main/java/com/exadel/discount/controller/AuthenticationController.java

```diff
@@ -0,0 +1,61 @@
+package com.exadel.discount.controller;
+
+import com.exadel.discount.model.http.AuthenticationRequest;
+import com.exadel.discount.model.http.AuthenticationResponse;
+import com.exadel.discount.service.UserDetailsServiceImpl;
+import com.exadel.discount.util.jwt.JwtUtil;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.http.ResponseEntity;
+
+@RestController
+@RequestMapping("/authenticate")
+@RequiredArgsConstructor
+public class AuthenticationController {
+    private final AuthenticationManager authenticationManager;
+    private final UserDetailsServiceImpl userDetailsService;
+    private final JwtUtil accessJwtUtil;
+    private final JwtUtil refreshJwtUtil;
+
+    @PostMapping("/login")
+    public ResponseEntity<?> createAuthenticationToken(@RequestBody AuthenticationRequest request) throws Exception {
+        try {
+            authenticationManager.authenticate(
+                    new UsernamePasswordAuthenticationToken(
+                            request.getUsername(),
+                            request.getPassword()
+                    ));
+        } catch (AuthenticationException ex) {
```

Comment:
Don't catch it here. Move it to the global exception handling. Ask @el-dub on details.
After moving this you'll not need ResponseEntity any more here as well

Reply:
so change it!